src/schema/mod.rs

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Commands to manage a telemetry schema.
+
+mod json_schema;
+mod resolve;
+
+use crate::schema::json_schema::SchemaJsonSchemaArgs;
+use crate::schema::resolve::SchemaResolveArgs;
+use crate::CmdResult;
+use clap::{Args, Subcommand};
+use weaver_cache::Cache;
+use weaver_common::Logger;
+
+/// Parameters for the `registry` command
+#[derive(Debug, Args)]
+pub struct SchemaCommand {
+    /// Define the sub-commands for the `schema` command
+    #[clap(subcommand)]
+    pub command: SchemaSubCommand,
+}
+
+/// Sub-commands to manage a `schema`.
+#[derive(Debug, Subcommand)]
+#[clap(verbatim_doc_comment)]
+pub enum SchemaSubCommand {
+    /// Resolves a telemetry schema.
+    ///
+    /// Rego policies present in the registry or specified using -p or --policy will be automatically validated by the policy engine before the artifact generation phase.
+    ///
+    /// Note: The `-d` and `--registry-git-sub-dir` options are only used when the registry is a Git URL otherwise these options are ignored.
+    ///
+    /// The process exits with a code of 0 if the resolution is successful.
+    #[clap(verbatim_doc_comment)]
+    Resolve(SchemaResolveArgs),
```

Comment:
I'm not sure I understand what this is meant to do.

Reply:
We need to discuss that in a more interactive way.

Reply:
Conversation initiated here https://cloud-native.slack.com/archives/C0697EXNTL3/p1717183349294419?thread_ts=1717104224.284179&cid=C0697EXNTL3